src/select/index.tsx

```diff
@@ -139,7 +139,7 @@ export const Select = factory(function Select({
 		data
 	} = get(options(), { read, meta: true });
 
-	if (required && dirty) {
+	if ((required && dirty) || value !== undefined) {
```

Comment:
The extra parenthesis shouldn't be necessary
```suggestion
	if (required && dirty || value !== undefined) {
```